app/models/project.server.ts

```diff
@@ -197,121 +210,143 @@ export async function getProjects(where: ProjectWhereInput) {
-      return { id: r.id }
-    })
+  for (let i = 0; i < projectMembers.length; i++) {
     // Create only the members that don't exist in this project
-    if (data.projectMembers[index].profile) {
-      activeMembers.push(data.projectMembers[index]?.id)
+    const previousMember = previousMembers.find(
+      (element) => element.profileId == projectMembers[i].profileId
+    );
+    if (previousMember !== undefined) {
+      activeMembers.push(projectMembers[i].profileId);
       // Just disconnects ALL related practicedSkills and roles, so it can UPDATE just the new selected ones after...
-      await db.projectMembers.update({
-        where: { id: data.projectMembers[index].id },
-        data: {
-          practicedSkills: { set: [] },
-          role: { set: [] },
-        },
-        include: {
-          practicedSkills: true,
-          role: true,
-        },
-      })
+      await db.$queryRaw`DELETE FROM "_ProjectMembersToSkills" WHERE "A" = ${previousMember.id}`;
+      await db.$queryRaw`DELETE FROM "_DisciplinesToProjectMembers" WHERE "B" = ${previousMember.id}`;
       // Makes all the actual updates to the projectMember
       await db.projectMembers.update({
-        where: { id: data.projectMembers[index].id },
+        where: { id: previousMember.id },
         data: {
-          hoursPerWeek: data.projectMembers[index].hoursPerWeek,
-          role: { connect: rolesArrayConnect },
-          active: data.projectMembers[index].active,
-          practicedSkills: { connect: practicedSkillsArrayConnect },
+          hoursPerWeek: projectMembers[i].hoursPerWeek,
+          role: { connect: projectMembers[i].role },
+          active: projectMembers[i].active,
+          practicedSkills: { connect: projectMembers[i].practicedSkills },
         },
         include: {
           practicedSkills: true,
         },
-      })
+      });
     } else {
       await db.projectMembers.create({
         data: {
-          project: { connect: { id } },
-          profile: { connect: { id: data.projectMembers[index].profileId } },
-          hoursPerWeek: data.projectMembers[index].hoursPerWeek,
-          role: { connect: rolesArrayConnect },
-          practicedSkills: { connect: practicedSkillsArrayConnect },
+          project: { connect: { id: projectId } },
+          profile: { connect: { id: projectMembers[i].profileId } },
+          hoursPerWeek: projectMembers[i].hoursPerWeek,
+          role: { connect: projectMembers[i].role },
+          practicedSkills: { connect: projectMembers[i].practicedSkills },
         },
-      })
+      });
     }
   }
 
-  // Delete members who are no longer active
-  for (let j = 0; j < data.existedMembers?.length; j++) {
-    if (!activeMembers.includes(data.existedMembers[j])) {
-      await db.projectMembers.deleteMany({ where: { id: data.existedMembers[j] } })
+  // Delete previous members who are no longer in activeMembers array of ids
+  for (let j = 0; j < previousMembers.length; j++) {
```

Comment:
Same here, we could extract the deletion into a function and create a function that performs the Deletion with a descriptive name